shell/platform/android/io/flutter/embedding/engine/systemchannels/LifecycleChannel.java

```diff
@@ -15,23 +16,69 @@
  */
 public class LifecycleChannel {
 
+  public enum AppLifecycleState { INACTIVE, RESUMED, PAUSED }
+
   @NonNull
   public final BasicMessageChannel<String> channel;
 
+  /**
+   * Cache the current state so that any queries for it may access it.
+   */
+  private AppLifecycleState state = AppLifecycleState.INACTIVE;
+
   public LifecycleChannel(@NonNull DartExecutor dartExecutor) {
     this.channel = new BasicMessageChannel<>(dartExecutor, "flutter/lifecycle", StringCodec.INSTANCE);
+    this.channel.setMessageHandler(new LifecycleQueryMessageHandler(this));
+  }
+
+  public AppLifecycleState getCurrentState() {
+    return state;
+  }
+
+  public void sendCurrentState() {
+    channel.send(lifecycleStateToString(state));
   }
 
   public void appIsInactive() {
-    channel.send("AppLifecycleState.inactive");
+    state = AppLifecycleState.INACTIVE;
+    sendCurrentState();
   }
 
   public void appIsResumed() {
-    channel.send("AppLifecycleState.resumed");
+    state = AppLifecycleState.RESUMED;
+    sendCurrentState();
   }
 
   public void appIsPaused() {
-    channel.send("AppLifecycleState.paused");
+    state = AppLifecycleState.PAUSED;
+    sendCurrentState();
   }
 
+  public static String lifecycleStateToString(AppLifecycleState state) {
```

Comment:
Please add this to the enum definition, similar to the way all other channels with enums serialize and deserialize.